src/realm/array.cpp

```diff
@@ -1168,62 +1168,75 @@ void Array::update_width_cache_from_header() noexcept
 }
 
 // This method reads 8 concecutive values into res[8], starting from index 'ndx'. It's allowed for the 8 values to
-// exceed array length; in this case, remainder of res[8] will be left untouched.
+// exceed array length; in this case, remainder of res[8] will be be set to 0.
 template <size_t w>
 void Array::get_chunk(size_t ndx, int64_t res[8]) const noexcept
 {
     REALM_ASSERT_3(ndx, <, m_size);
 
-    // To make Valgrind happy. Todo, I *think* it should work without, now, but if it reappears, add memset again.
-    // memset(res, 0, 8*8);
-
-    if (REALM_X86_OR_X64_TRUE && (w == 1 || w == 2 || w == 4) && ndx + 32 < m_size) {
```

Comment:
This was limited to x86 because unaligned loads work there, but it's illegal c++ regardless and loading byte per byte gets most of the speed benefit anyway.